### PR TITLE
Feature/plot state

### PR DESCRIPTION
**Pull request by BaxterEaves**
- Added state.plot in pylace to plot PCC states
- FIXED indexing bug in pylace.Engine which would return rows in a random order. The order seemed to be fixed on engine creation, which indicates to me that it was tied to some sort of random state used in the hashmap.

### CODE REVIEW

**Review by schmidmt — APPROVED** [review]
The code is good. Just a question and a cleanup.

File: pylace/lace/plot.py [hunk 1]
```diff
@@ -209,6 +210,307 @@ def prediction_uncertainty(
...
+    ...    show_columns=False,
+    ...    ax=ax,
+    ...    cmap="YlGnBu",
+    ...    aspect="auto"
+    ... )
+    >>> _ = plt.axis("off")
+    >>> plt.show()
+    """
+    if ax is None:
+        ax = plt.gca()
```
[the remark this answers is not in the document]
Comment (BaxterEaves) [reply pylace/lace/plot.py:397]: Maybe. I just do [what seaborn does](https://github.com/mwaskom/seaborn/blob/f0136337211854b2e63e5d26bee8b12ab60e0f42/seaborn/categorical.py#L1718-L1720)

[the remark this answers is not in the document]
Comment (schmidmt) [reply pylace/lace/plot.py:397]: Seems reasonable then 👍

**Review by Swandog — APPROVED** [review]
I skimmed it, mostly for documentation. LGTM!

**Pull request merged**